clippy_lints/src/methods/mod.rs

```diff
@@ -3286,6 +3289,26 @@ fn lint_single_char_push_string(cx: &LateContext<'_>, expr: &hir::Expr<'_>, args
     }
 }
 
+/// lint for length-1 `str`s as argument for `insert_str`
+fn lint_single_char_insert_string(cx: &LateContext<'_>, expr: &hir::Expr<'_>, args: &[hir::Expr<'_>]) {
+    let mut applicability = Applicability::MachineApplicable;
+    if let Some(extension_string) = get_hint_if_single_char_arg(cx, &args[2], &mut applicability) {
+        let base_string_snippet =
+            snippet_with_applicability(cx, args[0].span.source_callsite(), "_", &mut applicability);
+        let pos_arg = snippet(cx, args[1].span, "..");
+        let sugg = format!("{}.insert({}, {})", base_string_snippet, pos_arg, extension_string);
+        span_lint_and_sugg(
+            cx,
+            SINGLE_CHAR_PUSH_STR,
```

Comment:
I think we should rename the lint now that it also includes `insert_str`.

There seem to be precedents for when the lint includes more cases:
https://github.com/rust-lang/rust-clippy/blob/cc1998f7b3cc04c891f92c62a52c0c45fa4c145a/clippy_lints/src/lib.rs#L1881-L1899